packages/Teams/dotnet/Actions/SendMessageToTeamsChannel.cs

```diff
@@ -119,20 +120,48 @@ public override async Task<DialogTurnResult> BeginDialogAsync(DialogContext dc,
                 teamsChannelId = dc.Context.Activity.TeamsGetChannelId();
             }
 
-            // Retrieve the bot appid from TurnState's ClaimsIdentity
-            string appId; 
-            if (dc.Context.TurnState.Get<ClaimsIdentity>(BotAdapter.BotIdentityKey) is ClaimsIdentity botIdentity)
+            Tuple<ConversationReference, string> result;
+
+            // Check for legacy adapter
+            if (dc.Context.Adapter is BotFrameworkAdapter botFrameworkAdapter)
             {
-                appId = JwtTokenValidation.GetAppIdFromClaims(botIdentity.Claims);
+                // TeamsInfo.SendMessageToTeamsChannelAsync requires AppCredentials
+                var credentials = dc.Context.TurnState.Get<IConnectorClient>()?.Credentials as MicrosoftAppCredentials;
+                if (credentials == null)
+                {
+                    throw new InvalidOperationException($"Missing credentials as {nameof(MicrosoftAppCredentials)} in {nameof(IConnectorClient)} from TurnState");
+                }
+
+                // The result comes back as a tuple, which is used to set the two properties (if present).
+                result = await TeamsInfo.SendMessageToTeamsChannelAsync(dc.Context, activity, teamsChannelId, credentials, cancellationToken: cancellationToken).ConfigureAwait(false);
             }
             else
```

Comment:
nit: you might want to have a specific elseif for CloudAdapter and then the else for an exception saying we don't know how to process it.

Reply:
Good point. fixed :)